modules/oe_translation_remote/src/Controller/RemoteContentTranslationPreviewController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_translation_remote\Controller;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\PageCache\ResponsePolicy\KillSwitch;
+use Drupal\oe_translation\Controller\ContentTranslationPreviewController;
+use Drupal\oe_translation\Entity\TranslationRequestInterface;
+use Drupal\oe_translation\TranslationSourceManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Controller for previewing remote content translation request.
+ */
+class RemoteContentTranslationPreviewController extends ContentTranslationPreviewController {
+
+  /**
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * Creates a RemoteContentTranslationPreviewController object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\PageCache\ResponsePolicy\KillSwitch $killSwitch
+   *   The page cache kill switch.
+   * @param \Drupal\oe_translation\TranslationSourceManagerInterface $translationSourceManager
+   *   The translation source manager.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, KillSwitch $killSwitch, TranslationSourceManagerInterface $translationSourceManager, LanguageManagerInterface $language_manager) {
+    parent::__construct($entity_type_manager, $killSwitch, $translationSourceManager);
+    $this->languageManager = $language_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('page_cache_kill_switch'),
+      $container->get('oe_translation.translation_source_manager'),
+      $container->get('language_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function previewRequest(TranslationRequestInterface $oe_translation_request, string $language): array {
+    // Generate the translation preview for the request in the current language.
+    $language = $this->languageManager->getCurrentLanguage()->getId();
+    $translation = $this->getTranslationPreview($oe_translation_request, $language, $oe_translation_request->bundle());
```

Comment:
You are passing the `$oe_translation_request` entity, you don't need a separate argument for the bundle, you can just get it from the entity/